src/ts/lib/util/common.ts

```diff
@@ -945,6 +947,75 @@ class UtilCommon {
 		return String(s || '').replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
 	};
 
+	getPrismLanguageCodes () {
+		return Object.keys(components.languages).filter(x => !IGNORED_LANGUAGES.includes(x));
+	};
+
+	getPrismComponentRequirements (lang: string) {
+		let result = [];
+		const required = components.languages[lang].require || [];
+		if (Array.isArray(required)) {
+			for (const requiredLang of required) {
+				result = result.concat(this.getPrismComponentRequirements(requiredLang));
+			};
+		} else {
+			result = result.concat(this.getPrismComponentRequirements(required));
+		};
+		const optionals = components.languages[lang].optional || [];
+		if (Array.isArray(optionals)) {
+			for (const optional of optionals) {
+				result = result.concat(this.getPrismComponentRequirements(optional));
+			};
+		} else {
+			result = result.concat(this.getPrismComponentRequirements(optionals));
+		};
+		result.push(lang);
+		return result;
+	};
+
+	getPrismLanguagesAndAliases () {
+		const result = {};
```

Comment:
This logic seems very complicated, could you clarify what is going on here?

Reply:
Why not create global map with titles in the constructor? You create map inside the component and pass it as an argument into method, this is not very good practice:

```
class UtilCommon {

	prismMap = {};

	constructor () {
		this.prismMap = this.prismAliasMap();
	};

prismAliasMap () {
		const components = require(`prismjs/components.js`);
		const codes = components.languages || {};
		const result = {};

		Object.keys(codes).forEach((key: string) => {
			const lang = components.languages[key];
			const alias = [].concat(lang.alias || []);
			const titles = lang.aliasTitles || {};

			if (lang.title) {
				result[key] = lang.title;
			};

			if (alias.length) {
				alias.forEach((a: string) => {
					result[a] = titles[a] || lang.title;
				});
			};
		});

		return result;
	};
```

Reply:
No need to check for `meta` as well as it does not have title.